### PR TITLE
Improvement/vltclt 37 remove arsenal dep

### PR DESCRIPTION
**Pull request by ghost**
Remove arsenal from the prod dependencies, as it is only used to check errors (internal error, mostly), that can be stored as a const.
The main advantage being the decrease of the build duration, arsenal requires to build multiple modules, taking up to 5~10min every time in some CIs, depending on the load.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,13 +14,15 @@
   },
   "homepage": "https://github.com/scality/vaultclient#readme",
   "dependencies": {
-    "arsenal": "git+https://github.com/scality/Arsenal#8.1.91",
+    "@aws-crypto/sha256-universal": "^2.0.1",
+    "@aws-sdk/signature-v4": "^3.130.0",
```
Comment (tmacro) [inline package.json]: NPM is indicating that this package has been moved to `@smithy/signature-v4`

File: lib/IAMClient.js [hunk 2]
```diff
@@ -1014,7 +1021,7 @@ class VaultClient {
      * @param {string} [contentType] - content type of body
      * @returns {undefined}
      */
-    request(method, path, iamAuthenticate, callback, data, reqUid,
+    async request(method, path, iamAuthenticate, callback, data, reqUid,
```
Comment (tmacro) [inline lib/IAMClient.js]: Adding the `async` tag changes the return to a Promise, which in the case of `signer.sign` throwing would result in a `UnhandlePromiseRejection` and crash the process.

I'd suggest adding a `_signRequest` helper that is async and then using `then` and `catch` in request.
Something like:

```
async _signRequest(iamAuthenticate, req, options) 
```

That could then be used in request like
```
this._signRequest(iamAuthenticate, req, options)
    .catch(callback)
    .then(() => {
          ...
    });
```

Comment (ghost) [reply lib/IAMClient.js]: The helper was created, and the logic structure was reworked a bit to keep a flat function:
https://github.com/scality/vaultclient/pull/431/commits/1ce56261a91fa870df09cf0af9be14f1b7f12356

File: package.json (hunk 1 above)
Comment (ghost) [reply package.json]: Done here: https://github.com/scality/vaultclient/pull/431/commits/9fe22ac2e4bfb8341b5b58aa5e97863818cdb83f

**Review by KazToozs — APPROVED** [review]
lgtm

**Review by tmacro — APPROVED** [review]

**Pull request merged**